packages/ui/ImageItem/ImageItem.js

```diff
@@ -0,0 +1,243 @@
+
+		/**
+		 * Applies a selected visual effect to the image, but only if `selectionOverlayShowing`
+		 * is also `true`.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		selected: PropTypes.bool,
+
+		/**
+		 * The custom selection overlay component to render.
+		 *
+		 * A component can be a stateless functional component, `kind()` or React component.
+		 * The following is an example with custom selection overlay kind.
+		 *
+		 * Example:
+		 * ```
+		 * const SelectionOverlay = kind({
+		 * 	render: () => <div>custom overlay</div>
+		 * });
+		 *
+		 * <ImageItem selectionOverlay={SelectionOverlay} />
+		 * ```
+		 *
+		 * @type {Function}
+		 * @public
+		 */
+		selectionOverlay: PropTypes.func,
+
+		/**
+		 * Shows a selection overlay with a centered icon. When `selected` is true, a check mark is shown.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		selectionOverlayShowing: PropTypes.bool,
+
+		/**
+		 * The absolute URL path to the image.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		source: PropTypes.string,
+
+		/**
+		 * The second caption line to be displayed with the image.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		subCaption: PropTypes.string,
+
+		/**
+		 * The components that will be shown near the image, aligned with `orientation`.
+		 *
+		 * @type {Array|Element}
+		 * @public
+		 */
+		subComponents: PropTypes.oneOfType([PropTypes.array, PropTypes.element])
+	},
+
+	defaultProps: {
+		captionComponent: 'div',
+		iconComponent: Icon,
+		imageComponent: Image,
+		orientation: 'vertical',
+		selected: false,
+		selectionOverlayShowing: false
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'imageItem',
+		publicClassNames: true
+	},
+
+	computed: {
+		className: ({caption, selected, styler, subCaption}) => styler.append(
+			{selected, caption, subCaption}
+		),
+		selectionOverlay: ({css, iconComponent: IconComponent, selectionOverlay: SelectionOverlay, selectionOverlayShowing}) => {
+			if (selectionOverlayShowing) {
+				return (
+					<div className={css.overlayContainer}>
+						{
+							SelectionOverlay ?
+								<SelectionOverlay /> :
+								<div className={css.overlayComponent}>
+									<IconComponent className={css.icon}>check</IconComponent>
+								</div>
+						}
+					</div>
+				);
+			}
+		},
+		subComponents: ({caption, captionComponent: Caption, css, subCaption, subComponents}) => {
+			return (
+				subComponents ? subComponents : <React.Fragment>
+					{caption ? (<Cell className={css.caption} component={Caption} shrink>{caption}</Cell>) : null}
+					{subCaption ? (<Cell className={css.subCaption} component={Caption} shrink>{subCaption}</Cell>) : null}
+				</React.Fragment>
+			);
+		}
+	},
+
+	render: ({className, css, imageComponent: ImageComponent, orientation, placeholder, source, selectionOverlay, subComponents, ...rest}) => {
+		const classes = classNames(className, orientation === 'vertical' ? css.vertical : css.horizontal);
```

Comment:
Why don't you manipulate the class names in the `computed` block in a kind?
```
	computed: {
		className: ({orientation, styler}) => styler.append({vertical: , horizontal: })
	},
```

Reply:
oh yeah, good point.